ts-rs/src/lib.rs

```diff
@@ -423,15 +421,15 @@ pub trait TS {
     fn inline_flattened() -> String;
 
     /// Returns a [`TypeList`] of all types on which this type depends.
-    fn dependency_types() -> impl TypeList
+    fn visit_dependencies(_: &mut impl TypeVisitor)
```

Comment:
I imagine this entirely kills the recursion problem caused by `TypeList` right?

Reply:
Jup, exactly! There's no huge tuple anymore for which the compiler has to figure out if it implements `TypeList`.